core/src/main/java/org/apache/carbondata/core/datastore/page/ColumnPage.java

```diff
@@ -271,6 +267,18 @@ private static ColumnPage newIntPage(TableSpec.ColumnSpec columnSpec, int[] intD
     return columnPage;
   }
 
+  private static ColumnPage newBooleanPage(TableSpec.ColumnSpec columnSpec, byte[] booleanData) {
+    ColumnPage columnPage = createPage(columnSpec, BOOLEAN, booleanData.length);
```

Comment:
To make it simple, I think it is better to store boolean as byte in ColumnPage, so you can create a ColumnPage of Byte type.
When come to encoding, you still can do proper encoding of this column page.

Reply:
ok, I use newBytePage to replace newBooleanPage